File: spring-web/src/main/java/com/example/springwebex/model/pagenation/RequestBody.java
package com.example.springwebex.model.pagenation;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

import lombok.Data;

@Data
@JsonInclude(Include.NON_NULL)  // serialize
@JsonIgnoreProperties(ignoreUnknown = true) // deserialize
public class RequestBody {
    List<Integer> assetStateCodeList;
    Integer pageNumber;
    Integer pageSize;
}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/model/pagenation/RequestBody.kt
package com.example.springwebex.model.pagenation

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
data class RequestBody(
    var assetStateCodeList: List<Int>? = null,
    var pageNumber: Int? = null,
    var pageSize: Int? = null
)
